### ISSUE TITLE
ComboBox doesn't clear the filter text if you clear after setting new items

### ISSUE THREAD

**Issue by Prototailz** [vaadin/framework#9775]
ComboBox (with some initial items) doesn't clear the filter text if you use clear() right after setting new items. However, if you use setValue("") (with emptySelectionAllowed=true) instead of clear() or set clear() before setting new items it works as expected.

My project uses Vaadin 8.1.0 with the Vaadin CDI addon.

Here's some example code I used to verify this:

```java
public class ComboBoxTestView extends VerticalLayout implements View
{
   private static final long serialVersionUID = 1L;
   
   // Initial dataset
   private Collection<String> dataset = Arrays.asList("apple", "banana", "coconut", "date");
   private Collection<String> anotherDataset = Arrays.asList("azure", "blue", "cyan", "denim");
   private boolean datasetSwitch = false;
   
   private Button setValueButton;
   private Button clearBeforeButton;
   private Button clearAfterButton;
   private ComboBox<String> combobox;
   
   public ComboBoxTestView()
   {
      // This one works.
      setValueButton = new Button("Switch stuff & set to empty", e -> {
         switchDataset();
         combobox.setValue("");
      });
      
      // This one works too.
      clearBeforeButton = new Button("Clear & switch stuff", e -> {
         combobox.clear();
         switchDataset();
      });
      
      // This one doesn't work.
      clearAfterButton = new Button("Switch stuff & clear", e -> {
         switchDataset();
         combobox.clear();
      });
      
      combobox = new ComboBox<String>("Stuff", dataset);
      combobox.setEmptySelectionAllowed(true);
      
      addComponents(setValueButton, clearBeforeButton, clearAfterButton, combobox);
   }
   
   private void switchDataset()
   {
      if (datasetSwitch) {
         combobox.setItems(dataset);
      }
      else {
         combobox.setItems(anotherDataset);
      }
      
      datasetSwitch = !datasetSwitch;
   }
}
```

Steps to reproduce:
- select an item from the combobox
- click the third button
- verify that the combobox items have changed & the filter text doesn't match any of the items

**Comment by m-bermel** [comment]
Can confirm that on Vaadin 8.1.5. the ComboBox filter is not cleared after using clear(); or Binder.setBean(null);

**Comment by simon-chromow** [comment]
Duplication of #9047 and #9566

**Comment by TatuLund** [comment]
This is likely to be duplicate of https://github.com/vaadin/framework/issues/10284 

Have you checked 8.3.1, which has this fix: https://github.com/vaadin/framework/issues/10284 ? There will be further improvement in 8.4.0 via https://github.com/vaadin/framework/pull/10606

**Comment by simon-chromow** [comment]
From my point of view this is not a duplicate of #10284 but a duplicate of #9047 and  #9566

Still reproducible in 8.3.3:

The essential code part would be:
```java
// This one doesn't work.
clearAfterButton = new Button("Switch stuff & clear", e -> {
			switchDataset();
			combobox.clear();
		});
```

So if `combobox.setItems()` is called before `combobox.clear()` the filter text is not cleared.

**Comment by tsuoanttila** [comment]
Closing as duplicate of #9047. Seems to be fixed with my PR #10796
